lib/modules/versioning/composer/index.ts

```diff
@@ -67,6 +67,17 @@ function normalizeVersion(input: string): string {
   return convertStabilityModifier(output);
 }
 
+/**
+ * @param intput Version in any format, it recognizes the specific patch format x.x.x-pXX
+ * @returns If input contains the specific patch, it returns the input with removed the patch and true, otherwise it retunrs the same string and false.
+ */
+function removeComposerSpecificPatchPart(input: string): [string, boolean] {
+  const pattern = regEx(/-p[1-9][0-9]*$/gi);
```

Comment:
this would also match `2-beta1-p1` which should not.

why case insensitive?

Reply:
php seems to be case insensitive as well: https://github.com/composer/semver/blob/fa1ec24f0ab1efe642671ec15c51a3ab879f59bf/src/VersionParser.php#L137

in our case (Magento) - we retrieve the patch in lower-case, so it would work for us, but I am not sure if that would cover all cases. Should I remove the `i`?

Reply:
I fixed the issue related to matching beta1-p2